stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt

```diff
@@ -59,7 +63,10 @@ internal class ChatSocketServiceImpl constructor(
         }
 
         override fun onDisconnected() {
-            state = State.NetworkDisconnected
+            healthMonitor.stop()
+            if (state is State.Connected || state is State.Connecting) {
```

Comment:
When `Network` is disconnected, we should only update the `state` if it was `Connected` or `Connecting` into another case, we can keep the previous state